### ISSUE TITLE
published collection with MONGO idGeneration returns "Match failed" 400 error

### ISSUE THREAD

**Issue by yongwen** [stubailo/meteor-rest#41]
The /publications/<collection> url returns fine with the default meteor collection, but when I add the {idGeneration: 'MONGO'} option when creating the Collection, the url returns the following:
{
  "error": 400,
  "reason": "Match failed"
}
The collection example is:
Sensors = new Mongo.Collection('sensors', {idGeneration: 'MONGO'});
Meteor.publish('sensors', function() { return Sensors.find(); });
and the url is:
http://localhost:3000/publications/sensors

Looks like somehow the code did not handle the Mongo ObjectID correctly. any workaround or fix?

**Comment by aldeed** [comment]
Will need to change checks like [these](https://github.com/stubailo/meteor-rest/blob/devel/packages/rest/http-subscription.js#L25) to match on String or Mongo.ObjectId, and I guess convert it to a string for use as property name.

**Comment by stubailo** [comment]
Oops. That's definitely a bug!

**Comment by gerulf** [comment]
I also experienced this bug when I used simple:rest in an application that accesses a database created in Node.js using Mongoose.
As a solution I propose to replace check(id, String); in http-subscription.js by the following code:
  	if ( Match.test(id, Meteor.Collection.ObjectID) ) {
  		id = id + '';
  	}
  	check(id, String);  
So first a check is done if id has type ObjectId, if so it is converted to String by adding an empty string. Then check(id) is performed
With this modification my application runs without problems.
